src/main/java/org/openrewrite/gradle/RewriteRecipeAuthorAttributionTask.java

```diff
@@ -129,67 +145,151 @@ static class Attribution {
+                continue;
+            }
+
+            try {
                 // git commands only accept forward slashes in paths, regardless of operating system
                 String relativePath = getProject().getRootDir().toPath()
-                        .relativize(recipeFile.toPath()).toString()
+                        .relativize(yamlFile.toPath()).toString()
                         .replace('\\', '/');
-                List<Contributor> contributors = listContributors(g, relativePath);
+                BlameResult blame = g.blame().setFilePath(relativePath).call();
 
-                Attribution attribution = new Attribution("specs.openrewrite.org/v1beta/attribution", recipeFqn, contributors);
-                String yaml = mapper.writeValueAsString(attribution);
+                List<RecipeLineNumbers> recipeLineNumbers = recipeLineNumbers(yamlFile);
+                for (RecipeLineNumbers recipe : recipeLineNumbers) {
+                    List<Contributor> contributors = extractContributorsForRange(blame, recipe.startLine, recipe.endLine);
 
-                Files.createDirectories(targetPath.getParent());
-                Files.write(targetPath, yaml.getBytes());
-            }
+                    Attribution attr = new Attribution("specs.openrewrite.org/v1beta/attribution", recipe.name, contributors);
+                    String yaml = mapper.writeValueAsString(attr);
 
-        } catch (Exception e) {
-            getLogger().warn("Unable to complete attribution of recipe authors", e);
+                    Path targetPath = outputDir.resolve(recipe.name + ".yml");
+                    Files.createDirectories(targetPath.getParent());
+                    Files.writeString(targetPath, yaml);
+                }
+            } catch (Exception e) {
+                getLogger().warn("Unable to process YAML recipe file: " + yamlFile, e);
+            }
         }
     }
 
-    private static List<Contributor> listContributors(Git g, String relativeUnixStylePath) throws GitAPIException {
-        Map<Contributor, Integer> contributors = new HashMap<>();
+    record RecipeLineNumbers(String name, int startLine, int endLine) {
+    }
 
-        BlameResult blame = g.blame().setFilePath(relativeUnixStylePath).call();
-        if (blame == null) {
-            return Collections.emptyList();
+    private static List<RecipeLineNumbers> recipeLineNumbers(File file) throws IOException {
+        YAMLParser parser = new YAMLFactory().createParser(file);
+        var mapMappingIterator = new YAMLMapper()
+                .readValues(parser, new TypeReference<Map<String, Object>>() {
+                });
+
+        String recipeName = null;
+        int startLine = 0;
+        int endLine;
+        List<RecipeLineNumbers> list = new ArrayList<>();
+        while (mapMappingIterator.hasNext()) {
+            // Store the previous entry
+            endLine = mapMappingIterator.getCurrentLocation().getLineNr() - 1;
```

Comment:
`getLineNr()` is 1 based; whereas `getSourceAuthor(int)` used further down is 0 based. 🤷🏻